python/utils/LinkedLibraryHolder.cpp

```diff
@@ -30,28 +30,22 @@ constexpr static const char NVQIR_SIMULATION_BACKEND[] =
     "NVQIR_SIMULATION_BACKEND=";
 constexpr static const char TARGET_DESCRIPTION[] = "TARGET_DESCRIPTION=";
 
+/// @brief A utility function to check availability of Nvidia GPUs and return
+/// their count
 int countGPUs() {
-  char buffer[1024];
-  std::string output;
-  FILE *fp1, *fp2;
-
-  fp1 = popen("nvidia-smi", "r");
-  if (!fp1) {
+  auto retCode = std::system("nvidia-smi >/dev/null 2>&1");
+  if (0 != retCode) {
     cudaq::info("nvidia-smi: command not found");
-    return -1;
+    return 0;
   }
-  pclose(fp1);
 
-  fp2 = popen("nvidia-smi -L | wc -l", "r");
-  if (!fp2) {
-    cudaq::info("nvidia-smi: command not working");
-    return -1;
-  }
-  while (fgets(buffer, sizeof buffer, fp2)) {
-    output += buffer;
-  }
-  pclose(fp2);
-  return std::stoi(output);
+  std::string tmpFile = ".temp.command.capture.output.txt";
```

Comment:
Q: Alternate way to generate a temporary file name? `tmpnam` gives warning `warning: the use of 'tmpnam' is dangerous`

Reply:
The warning is hinting you should use `tmpnam_s` IIRC.

Reply:
Replaced the logic with `mkstemp`... is it better?

Reply:
Yes. `mkstemp` is even better. Thanks.